### PR TITLE
Allow KeyVault lookup plugin to continue when using on MSI-disabled Azure VM

### PR DESCRIPTION
**Pull request by MattT95**
This fix is to ensure that if the lookup plugin is called on an Azure VM that has MSI disabled, the Service Principal fallback can still be considered.

Before this fix, a KeyError is thrown by plugin when using the lookup on an MSI-disabled machine, due to getting the following response payload from the identity endpoint:
```{u'error_description': u'Identity not found', u'error': u'invalid_request'}```

This occurs even when the Service Principal fallback is set, meaning that this fallback is not even considered and prevents you using the plugin on any Azure VM, unless you are using MSI.

After this change, when a request is successful to the metadata identity endpoint but there is no token provided in the response (in the case of the response shown above), a message is printed to indicate this and the Service Principal fallback mechanism is then considered

### CODE REVIEW

**Comment by MyronFanQiu** [comment]
@MattT95 Thanks for the contribution! We need some time to review the PR. BTW, can you rebase or update the branch to the current master branch?

**Comment by MyronFanQiu** [comment]
@MattT95 Hi. This PR looks good for me. Please update the branch and it will re-trigger the CI check. Meanwhile, we can wait some time to see if there are other review opinions. If everything is ok, we can merge this one.

**Comment by MattT95** [comment]
@MyronFanQiu Thanks for checking. I have updated/rebased the branch as requested.

**Comment by MyronFanQiu** [comment]
shipit

File: lookup_plugins/azure_keyvault_secret.py [hunk 1]
```diff
@@ -97,8 +97,11 @@
 token = None
 try:
     token_res = requests.get('http://169.254.169.254/metadata/identity/oauth2/token', params=token_params, headers=token_headers)
-    token = token_res.json()["access_token"]
-    TOKEN_ACQUIRED = True
+    token = token_res.json().get("access_token")
+    if token is not None:
+        TOKEN_ACQUIRED = True
+    else:
+        print('Successfully called MSI endpoint, but no token was available. Will use service principal if provided.')
```
Comment (yungezz) [inline lookup_plugins/azure_keyvault_secret.py]: could you pls use ansible logger instead of  print?

**Comment by MyronFanQiu** [comment]
@MattT95 Hi, I checked some plugins and you can refer [this](https://github.com/ansible/ansible/blob/f7e5494a67ae2f5b8480ab6c0f125b8a30f4da28/lib/ansible/plugins/lookup/file.py#L69) to solve the review comments. Hope it will help!

**Comment by MattT95** [comment]
@MyronFanQiu @yungezz thanks for the comments and help. I'll look round to getting this done in the next couple of days :).

**Comment by MyronFanQiu** [comment]
@MattT95 Thanks for your contribution!

**Pull request merged**